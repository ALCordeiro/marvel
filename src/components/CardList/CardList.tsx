import React from 'react';

import { CardList } from './CardList.style';
import { CardComponent } from '../Card';

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export const CardListComponent = ({ characters }: any) => (
    <CardList>
        {characters.map((character: any) => (
            <CardComponent character={character} />
        ))}
    </CardList>
);

export default CardListComponent;
