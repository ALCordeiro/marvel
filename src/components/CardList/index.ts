export { default as CardListComponent } from './CardList';
