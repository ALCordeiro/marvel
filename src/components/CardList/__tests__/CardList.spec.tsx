/* eslint-disable no-undef */
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';
import { CardListComponent } from '../CardList';

const characters: any = [
    {
        comics: {
            available: 12,
            collectionURI:
                'http://gateway.marvel.com/v1/public/characters/1011334/comics',
            items: Array(12),
            returned: 12,
        },
        description: '',
        events: {
            available: 1,
            collectionURI:
                'http://gateway.marvel.com/v1/public/characters/1011334/events',
            items: Array(1),
            returned: 1,
        },
        id: 1011334,
        modified: '2014-04-29T14:18:17-0400',
        name: '3-D Man',
        resourceURI: 'http://gateway.marvel.com/v1/public/characters/1011334',
        series: {
            available: 3,
            collectionURI:
                'http://gateway.marvel.com/v1/public/characters/1011334/series',
            items: Array(3),
            returned: 3,
        },
        stories: {
            available: 21,
            collectionURI:
                'http://gateway.marvel.com/v1/public/characters/1011334/stories',
            items: Array(20),
            returned: 20,
        },
        thumbnail: {
            path: 'http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784',
            extension: 'jpg',
        },
        map: () => 'test',
        length: () => 1,
        filter: () => 'test',
    },
];

let wrapper;

beforeEach(() => {
    wrapper = render(
        <Router>
            <CardListComponent characters={characters} />
        </Router>,
    );
});

test('should render', async () => {
    const { container } = wrapper;
    expect(container.firstChild);
});

test('loads and displays correct card', async () => {
    expect(wrapper.getByRole('heading')).toHaveTextContent('3-D Man');
});

test('should match snapshot', () => {
    const { container } = wrapper;
    expect(container.firstChild).toMatchSnapshot();
});
