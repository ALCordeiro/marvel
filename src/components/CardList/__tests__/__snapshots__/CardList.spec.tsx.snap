// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  class="sc-bdnxRM iQMexA"
>
  <a
    href="/details?id=1011334"
    style="text-decoration: none;"
  >
    <div
      class="sc-gtsrHT bVkflj"
    >
      <div
        class="sc-dlnjwi dhkUjL"
      >
        <div
          class="sc-hKFxyN idpmax"
        />
      </div>
      <div
        class="sc-eCApnc bKvsvX"
      >
        <h1
          class="sc-jSFjdj dkaRHx"
        >
          3-D Man
        </h1>
      </div>
    </div>
  </a>
</div>
`;
